### ISSUE TITLE
parse<boolean> is identified as UINT8

### ISSUE THREAD

**Issue by Gabriele91** [microsoft/krabsetw#177]
Hi there,

I have a problem regarding parse<T> for boolean type.
Indeed it is identified as UINT8 instead of BOOLEAN, thus throw_if_invalid raises an exception:
![image](https://user-images.githubusercontent.com/641004/132472636-4530a2ec-9910-492e-8bd8-476b07b3fd53.png)

I guess it happens because in c++ boolean is an unsigned char instead of int. I think could be nice a specialized template for the boolean type, in order to avoid this problem.

**Comment by Gabriele91** [comment]
Update, It is also true for the BOOL type, which is classified as INT32 instead of BOOLEAN:
![image](https://user-images.githubusercontent.com/641004/132474264-896a56c6-16bb-4f56-92ee-2a95dfca67f6.png)
I guess a workaround is a struct krabs::boolean like the krabs::hex32/64  ones.

**Comment by Gabriele91** [comment]
This is my workaround and it works fine, for me.
```C++
namespace krabs
{
    struct boolean
    {
        int value;
    };

    namespace debug
    {
        template <>
        inline void assert_valid_assignment<boolean>(const std::wstring&, const property_info& info)
        {
            auto actual = (_TDH_IN_TYPE)info.pEventPropertyInfo_->nonStructType.InType;
            if (actual != TDH_INTYPE_BOOLEAN) {
                throw std::runtime_error("Requested an boolean value from non-boolean property");
            }
        }
    }
}
```

**Comment by kasandrk** [comment]
Use (BOOLEAN)parse<bool>

**Comment by kasandrk** [comment]
Use `(BOOLEAN)parse<bool>`

**Comment by Gabriele91** [comment]
Nope, the issue is visible in the first image, the specialized template classifies it as UINT8 instead of boolean (TDH_INTYPE_BOOLEAN). This raises the exception.
